sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsMeta.scala

```diff
@@ -447,13 +447,27 @@ abstract class SparkPlanMeta[INPUT <: SparkPlan](plan: INPUT,
     case _ => childPlans.flatMap(_.findShuffleExchanges())
   }
 
+  private def findBucketedReads(): Seq[Boolean] = wrapped match {
+    case f: FileSourceScanExec =>
+      if (f.bucketedScan) {
+        true :: Nil
+      } else {
+        false :: Nil
+      }
+    case _ =>
+      childPlans.flatMap(_.findBucketedReads())
```

Comment:
This code won't recurse down into query stages that have already executed (because query stages are leaf nodes), but I think that's ok since we would already have tagged the initial plan.

Reply:
The goal is to find the next closest shuffle or read.  If we encounter a shuffle before we hit a read it is fine, because the other code path should handle it.  We only need to sorry about an input in the same stage.